### PR TITLE
Spark.NET: new overview docs

### PR DESCRIPTION
**Pull request by mamccrea**
## Summary

Updated **What is .NET for Apache Spark?** and added **What is Apache Spark?**

Fixes #Issue_Number (if available)

### CODE REVIEW

File: docs/spark/toc.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@
 - name: Overview
   expanded: true
   items:
+  - name: What is Apache Spark?
+    href: what-is-spark.md
```
Comment (mairaw) [inline docs/spark/toc.yml]: nit: typically if you run the SEO tests on CATS, they will flag when you have duplicated words on the path.

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 2]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
 
 ## Contributions
 
 The .NET for Apache Spark team encourages contributions, both GitHub issues and pull requests. First, look for an [existing issue](https://github.com/dotnet/spark/issues). If you can't find an existing issue, [open a new issue](https://github.com/dotnet/spark/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+).
+
+## Next steps
+
+Try .NET for Apache Spark.
+> [!div class="nextstepaction"]
+> [Tutorial: Get started with .NET for Apache Spark](../tutorials/get-started.md)
```
Comment (mairaw) [inline docs/spark/what-is-apache-spark-dotnet.md]:
```suggestion
> [Tutorial: Get started with .NET for Apache Spark](./tutorials/get-started.md)
```

File: docs/spark/what-is-spark.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+description: Discover how to run a .NET for Apache Spark app using .NET Core on Windows.
+ms.date: 09/26/2019
+ms.topic: conceptual
+ms.custom: mvc
+#Customer intent: As a developer, I want an introduction to Apache Spark.
+---
+
+# What is Apache Spark?
+
+[Apache Spark](https://spark.apache.org/) is an open-source parallel processing framework that supports in-memory processing to boost the performance of applications that analyze big data. A Spark job can repeatedly query data that is loaded and cached into memory, which is much faster than disk-based alternatives. Spark is a general purpose distributed processing engine that can be used in many big data scenarios, including:
```
Comment (mairaw) [inline docs/spark/what-is-spark.md]:
```suggestion
[Apache Spark](https://spark.apache.org/) is an open-source parallel processing framework that supports in-memory processing to boost the performance of applications that analyze big data. A Spark job can repeatedly query data that is loaded and cached into memory, which is much faster than disk-based alternatives. Spark is a general-purpose distributed processing engine that can be used in many big data scenarios, including:
```

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 4]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
 .NET for Apache Spark runs on Windows, Linux, and macOS using .NET Core. It also runs on Windows using .NET Framework. You can deploy your applications to all major cloud providers including Azure HDInsight Spark, Amazon EMR Spark, Azure Databricks, and Databricks on AWS.
 
+## .NET for Apache Spark architecture
+
+The C#/ F# language binding to Spark is written on a new Spark interop layer which offers easier extensibility. This new layer of Spark interop was written using the best practices for language extension and optimizes for interop and performance. Long term this extensibility can be used for adding support for other languages in Spark.
+
+> [!div class="mx-imgBorder"]
+> ![.NET for Apache Spark architecture](media/dotnet-spark-architecture.png)
+
+You can about interop support for Spark language extensions from [the proposal](https://issues.apache.org/jira/browse/SPARK-26257).
```
Comment (mairaw) [inline docs/spark/what-is-apache-spark-dotnet.md]: You can learn?

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 5]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
+* Spark SQL
+* MLlib
 
 .NET for Apache Spark is compliant with .NET Standard, a formal specification of .NET APIs that are common across .NET implementations. This means you can use .NET for Apache Spark anywhere you write .NET code allowing you to reuse all the knowledge, skills, code, and libraries you already have as a .NET developer.
 
 .NET for Apache Spark runs on Windows, Linux, and macOS using .NET Core. It also runs on Windows using .NET Framework. You can deploy your applications to all major cloud providers including Azure HDInsight Spark, Amazon EMR Spark, Azure Databricks, and Databricks on AWS.
 
+## .NET for Apache Spark architecture
+
+The C#/ F# language binding to Spark is written on a new Spark interop layer which offers easier extensibility. This new layer of Spark interop was written using the best practices for language extension and optimizes for interop and performance. Long term this extensibility can be used for adding support for other languages in Spark.
```
Comment (mairaw) [inline docs/spark/what-is-apache-spark-dotnet.md]:
```suggestion
The C#/ F# language binding to Spark is written on a new Spark interop layer which offers easier extensibility. This new layer of Spark interop was written using the best practices for language extension and optimizes for interop and performance. Long term, this extensibility can be used for adding support for other languages in Spark.
```

**Review by mairaw — COMMENTED** [review]
Left some comments for you to consider.

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 6]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
 author: mamccrea
 ms.topic: overview
-ms.date: 05/06/2019
+ms.date: 09/27/2019
 ---
 
 # What is .NET for Apache Spark?
 
-[Apache Spark](https://spark.apache.org/) is a general-purpose distributed processing engine for analytics over large data sets-typically terabytes or petabytes of data. .NET for Apache Spark makes Apache Spark accessible for developers, bringing the power of Spark to your applications with support for languages you already know.
+[Apache Spark](what-is-spark.md) is a general-purpose distributed processing engine for analytics over large data sets - typically terabytes or petabytes of data. With .NET for Apache Spark, a free, open-source, and cross-platform big data analytics framework, you can now add the power of Apache Spark to your big data applications using languages you already know.
```
Comment (MikeRys) [inline docs/spark/what-is-apache-spark-dotnet.md]: I am torn here. .NET for Apache Spark is not the analytics framework per se (that is Apache Spark) but allows .NET users to use it. What about:

With .NET for Apache Spark, the free, open-source, and cross-platform .NET Support for the popular open-source big data analytics framework, you can now add the power of Apache Spark to your big data applications using languages you already know.

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 7]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
+The C#/ F# language binding to Spark is written on a new Spark interop layer which offers easier extensibility. This new layer of Spark interop was written using the best practices for language extension and optimizes for interop and performance. Long term this extensibility can be used for adding support for other languages in Spark.
+
+> [!div class="mx-imgBorder"]
+> ![.NET for Apache Spark architecture](media/dotnet-spark-architecture.png)
+
+You can about interop support for Spark language extensions from [the proposal](https://issues.apache.org/jira/browse/SPARK-26257).
+
+## .NET for Apache Spark performance
+
+When compared against Python and Scala using [TPC-H benchmark](http://www.tpc.org/tpch/), .NET for Apache Spark performs well in most cases and is 2x faster than Python when user-defined function performance is critical. There is an ongoing effort to improve and benchmark performance.
```
Comment (MikeRys) [inline docs/spark/what-is-apache-spark-dotnet.md]:
```suggestion
When compared against Python and Scala using the [TPC-H benchmark](http://www.tpc.org/tpch/), .NET for Apache Spark performs well in most cases: if no user-defined functions are present, it performs as fast as Scala and Python, and it is 2x faster than Python when user-defined function performance is critical. The .NET for Apache Spark project is continuing to improve and benchmark performance. 
```

**Review by MikeRys — CHANGES_REQUESTED** [review]
Left two suggestions.

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 8]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
 
-[Apache Spark](https://spark.apache.org/) is a general-purpose distributed processing engine for analytics over large data sets-typically terabytes or petabytes of data. .NET for Apache Spark makes Apache Spark accessible for developers, bringing the power of Spark to your applications with support for languages you already know.
+[Apache Spark](what-is-spark.md) is a general-purpose distributed processing engine for analytics over large data sets - typically terabytes or petabytes of data. With .NET for Apache Spark, a free, open-source, and cross-platform big data analytics framework, you can now add the power of Apache Spark to your big data applications using languages you already know.
 
-With C# and F#, you can access:
+## Why choose .NET for Apache Spark?
 
-* Dataframe and SparkSQL for working with structured data.
-* Spark Structured Streaming for working with streaming data.
+.NET for Apache Spark provides high performance APIs for using Spark from C# and F#. With C# and F#, you can access:
```
Comment (bamurtaugh) [inline docs/spark/what-is-apache-spark-dotnet.md]: The goal of .NET Spark is empowering developers with .NET knowledge/code bases to participate in the world of big data, and a non-goal is to convert devs with expertise in other languages to choose .NET instead. Thus, I think we'd want to emphasize this in this section titled "Why choose .NET for Apache Spark?" so that it doesn't come across that we're trying to convert devs from other languages. 

Maybe rephrasing this opening sentence/introducing a new sentence like:
".NET for Apache Spark empowers developers with .NET experience or code bases to participate in the world of big data analytics. .NET for Apache Spark provides high performance APIs..."

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 9]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
+
+* DataFrame and SparkSQL for working with structured data
+* Spark Structured Streaming for working with streaming data
+* Spark SQL
+* MLlib
```
Comment (bamurtaugh) [inline docs/spark/what-is-apache-spark-dotnet.md]: We don't currently support MLlib (although it may be on the road map). We might want to rephrase this as just machine learning, and could mention ML.NET integration:

```suggestion
* Machine learning integration for faster training and prediction (i.e. use .NET for Apache Spark alongside [ML.NET](http://dot.net/ml))
```

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 10]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
-With C# and F#, you can access:
+## Why choose .NET for Apache Spark?
 
-* Dataframe and SparkSQL for working with structured data.
-* Spark Structured Streaming for working with streaming data.
+.NET for Apache Spark provides high performance APIs for using Spark from C# and F#. With C# and F#, you can access:
+
+* DataFrame and SparkSQL for working with structured data
+* Spark Structured Streaming for working with streaming data
+* Spark SQL
```
Comment (bamurtaugh) [inline docs/spark/what-is-apache-spark-dotnet.md]: We'd probably want to more clearly differentiate Spark SQL (writing queries with SQL syntax) from SparkSQL (for working with structured data)

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 11]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
+[Apache Spark](what-is-spark.md) is a general-purpose distributed processing engine for analytics over large data sets - typically terabytes or petabytes of data. With .NET for Apache Spark, a free, open-source, and cross-platform big data analytics framework, you can now add the power of Apache Spark to your big data applications using languages you already know.
 
-With C# and F#, you can access:
+## Why choose .NET for Apache Spark?
 
-* Dataframe and SparkSQL for working with structured data.
-* Spark Structured Streaming for working with streaming data.
+.NET for Apache Spark provides high performance APIs for using Spark from C# and F#. With C# and F#, you can access:
+
+* DataFrame and SparkSQL for working with structured data
```
Comment (bamurtaugh) [inline docs/spark/what-is-apache-spark-dotnet.md]: For all of the scenarios in this list (structured/batch data, streaming, SQL, ML), it would be beneficial to provide some more explanation/context- when I was new to big data, I didn't know what any of this was, and it'd be great to explain it here so that users don't leave confused or have to search on the Internet for further definitions.

Maybe sub-bullets, an additional section, or just longer bullets defining these scenarios/data types: batch data, structured data, Spark SQL, ML training and prediction

File: docs/spark/what-is-apache-spark-dotnet.md [hunk 12]
```diff
@@ -3,26 +3,55 @@ title: What is .NET for Apache Spark?
...
+* Spark Structured Streaming for working with streaming data
+* Spark SQL
+* MLlib
 
 .NET for Apache Spark is compliant with .NET Standard, a formal specification of .NET APIs that are common across .NET implementations. This means you can use .NET for Apache Spark anywhere you write .NET code allowing you to reuse all the knowledge, skills, code, and libraries you already have as a .NET developer.
```
Comment (bamurtaugh) [inline docs/spark/what-is-apache-spark-dotnet.md]:
```suggestion
.NET for Apache Spark is compliant with .NET Standard, a formal specification of .NET APIs that are common across .NET implementations. This means you can use .NET for Apache Spark anywhere you write .NET code, allowing you to reuse all the knowledge, skills, code, and libraries you already have as a .NET developer.
```

File: docs/spark/what-is-spark.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+
+![Apache Spark architecture](media/spark-architecture.png)
+
+### Driver
+
+The driver consists of your program, like a C# console app, and a Spark session. The Spark session takes your program and divides it into smaller tasks that are handled by the executors.
+
+### Executors
+
+Each executor, or worker node, receives a task from the driver and executes that task.
```
Comment (bamurtaugh) [inline docs/spark/what-is-spark.md]:
```suggestion
Each executor, or worker node, receives a task from the driver and executes that task. The executors reside on an entity known as a cluster.
```

File: docs/spark/what-is-spark.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+- Spark Scala API
+- Spark Java API
+- Spark Python API
+- Spark R API
+- Spark SQL, built-in functions
+
+## Next steps
+
+Learn how you can use Apache Spark in your .NET application.
```
Comment (bamurtaugh) [inline docs/spark/what-is-spark.md]:
```suggestion
Up until now, there was not a .NET API for Apache Spark, locking .NET developers out of the world of big data analytics. With .NET for Apache Spark, developers with .NET experience and business logic can write big data queries in C# and F#. Learn how you can use Apache Spark in your .NET application.
```

File: docs/spark/what-is-spark.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+
+- Scala
+- Python
+- Java
+- SQL
+- R
+- .NET
+
+## Spark APIs
+
```
Comment (bamurtaugh) [inline docs/spark/what-is-spark.md]: Do we need separate sections for language support vs Spark APIs? Could these be combined/one of these sections removed? If there is a benefit to having these separate sections, I think it'd be beneficial to add a line or two introducing the Spark APIs section.

**Review by bamurtaugh — APPROVED** [review]
LGTM, thanks so much for working on this!

**Review by mairaw — APPROVED** [review]
